diesel/src/pg/types/array.rs

```diff
@@ -134,10 +134,13 @@ impl<'a, ST, T> ToSql<Array<ST>, Pg> for &'a [T] where
         let mut buffer = Vec::new();
         for elem in self.iter() {
             let is_null = try!(elem.to_sql(&mut buffer));
-            assert!(is_null == IsNull::No, "Arrays containing null are not supported");
-            try!(out.write_i32::<NetworkEndian>(buffer.len() as i32));
-            try!(out.write_all(&buffer));
-            buffer.clear();
+            if let IsNull::No = is_null {
+                try!(out.write_i32::<NetworkEndian>(buffer.len() as i32));
+                try!(out.write_all(&buffer));
+                buffer.clear();
+            } else {
+                try!(out.write_i32::<NetworkEndian>(-1));
```

Comment:
I'm not familiar with Postgres' binary interface, so this is one of the places where I'd really like to have a link to some semi-official Postgres document that says "-1 is null here".